src/components/Upload/UploadContext.tsx

```diff
@@ -0,0 +1,225 @@
+import React, {
+  createContext,
+  FC,
+  useCallback,
+  useContext,
+  useEffect,
+  useReducer,
+  useState,
+} from 'react';
+import { sleep } from '../../util/sleep';
+import * as actions from './Reducer/uploadActions';
+import { initialState } from './Reducer/uploadInitialState';
+import { uploadReducer } from './Reducer/uploadReducer';
+import * as Types from './Reducer/uploadTypings';
+import {
+  useDeleteFileNodeMutation,
+  useRequestFileUploadMutation,
+} from './Upload.generated';
+import { UploadItems } from './UploadItems';
+import { UploadManager } from './UploadManager';
+
+interface UploadContextValue {
+  addFilesToUploadQueue: (files: Types.FileInput[]) => void;
+}
+
+const initialUploadContext = {
+  addFilesToUploadQueue: () => null,
+};
+
+export const UploadContext = createContext<UploadContextValue>(
+  initialUploadContext
+);
+UploadContext.displayName = 'UploadContext';
+
+interface UploadManagerContextValue {
+  isManagerOpen: boolean;
+  setIsManagerOpen: (isOpen: boolean) => void;
+}
+
+const initialUploadManagerContext = {
+  isManagerOpen: false,
+  setIsManagerOpen: () => null,
+};
```

Comment:
Personally I don't like having to define the types again after defining the initial state. What I like to do is use the initial state to define the shape.
```suggestion
const initialUploadManagerContext = {
  isManagerOpen: false,
  setIsManagerOpen: (isOpen: boolean) => {
    // noop
  },
};
type UploadManagerContextValue = typeof initialUploadManagerContext;
```
Better still since the interface/type is only passed to the `createContext` function, it can just inferred the type from the initial state shape.

Really the trick is to make sure the initial state props declare the types correctly. Functions usually don't match so I've done this before too:
```tsx
  setIsManagerOpen: noop as (isOpen: boolean) => void
```

Reply:
Oh, interesting; I've never seen that before

Reply:
@CarsonF I tried this with the `noop` function from `ts-essentials`, and TypeScript yelled at me; were you using a different `noop` function?

Reply:
Umm lodash. But both should work. What's the problem?

Reply:
Nm, I figured it out:

`setIsManagerOpen: (_: boolean) => { return; },`